### PR TITLE
refactoring setup

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

refactoring setup to better support typing
Closes #6500

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by SeanNaren — APPROVED** [review]

File: pytorch_lightning/__init__.py [hunk 1]
```diff
@@ -50,32 +25,20 @@
...
 
-try:
-    # This variable is injected in the __builtins__ by the build
-    # process. It used to enable importing subpackages of skimage when
-    # the binaries are not built
-    _ = None if __LIGHTNING_SETUP__ else None
-except NameError:
-    __LIGHTNING_SETUP__: bool = False
-
-if __LIGHTNING_SETUP__:  # pragma: no-cover
```
Comment (awaelchli) [inline pytorch_lightning/__init__.py:61]: so you determined that this is not needed anymore? is it ok with the docs build?

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/__init__.py (hunk 1 above)
Comment (Borda) [reply pytorch_lightning/__init__.py:61]: doc seems to pass :]

Comment (awaelchli) [reply pytorch_lightning/__init__.py:61]: great!

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/setup_tools.py [hunk 2]
```diff
@@ -40,10 +40,10 @@ def _load_requirements(path_dir: str, file_name: str = 'requirements.txt', comme
     return reqs
 
 
-def _load_readme_description(path_dir: str, homepage: str = __homepage__, version: str = __version__) -> str:
+def _load_readme_description(path_dir: str, homepage: str, version: str) -> str:
```
Comment (carmocca) [inline pytorch_lightning/setup_tools.py:43]:
```suggestion
def _load_readme_description(path_dir: str, homepage: str = "", version: str = "") -> str:
```

File: pytorch_lightning/setup_tools.py [hunk 3]
```diff
@@ -40,10 +40,10 @@ def _load_requirements(path_dir: str, file_name: str = 'requirements.txt', comme
...
     """Load readme as decribtion
 
-    >>> _load_readme_description(_PROJECT_ROOT)  # doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
+    >>> _load_readme_description(_PROJECT_ROOT, "", "")  # doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
```
Comment (carmocca) [inline pytorch_lightning/setup_tools.py:46]:
```suggestion
    >>> _load_readme_description(_PROJECT_ROOT)  # doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
```

File: pytorch_lightning/setup_tools.py (hunk 2 above)
Comment (Borda) [reply pytorch_lightning/setup_tools.py:43]: why?

Comment (carmocca) [reply pytorch_lightning/setup_tools.py:43]: since you had to pass "" in the doctest and you are passing as keyword arguments in `setup.py`

**Pull request merged**